micropython/micropython-1.15-patchset/os.patch

```diff
@@ -261,13 +288,24 @@ diff -Naur micropython-1.15/ports/unix/Makefile micropython-phoenix/ports/unix/M
  
  INC +=  -I.
  INC +=  -I$(TOP)
-@@ -103,17 +103,9 @@
- endif
- LDFLAGS += $(LDFLAGS_MOD) $(LDFLAGS_ARCH) -lm $(LDFLAGS_EXTRA)
+ INC += -I$(BUILD)
  
+ # compiler settings
+-CWARN = -Wall -Werror
++CWARN = -Wall
+ CWARN += -Wextra -Wno-unused-parameter -Wpointer-arith -Wdouble-promotion -Wfloat-conversion
+ CFLAGS += $(INC) $(CWARN) -std=gnu99 -DUNIX $(CFLAGS_MOD) $(COPT) -I$(VARIANT_DIR) $(CFLAGS_EXTRA)
+ 
+@@ -101,19 +101,11 @@
+ # Use gcc syntax for map file
+ LDFLAGS_ARCH = -Wl,-Map=$@.map,--cref -Wl,--gc-sections
+ endif
+-LDFLAGS += $(LDFLAGS_MOD) $(LDFLAGS_ARCH) -lm $(LDFLAGS_EXTRA)
+-
 -# Flags to link with pthread library
 -LIBPTHREAD = -lpthread
--
++LDFLAGS = $(LDFLAGS_MOD) $(LDFLAGS_ARCH) -lm $(LDFLAGS_EXTRA)
```

Comment:
FAIL: changed `+=` to `=` and removed `--gc-sections` in the process

Also: 
 - don't remove `LIBPTHREAD` but set `MICROPY_PY_THREAD` correctly
 - don't remove sections of the Makefile, just patch the `mpconfigport.mk` as expected or even treat it as a config file (so add our own `mpconfigport.mk` to the repo and copy it over)

Edit:
I see that `mpconfigport.mk` is already changed, so half of the changes from this patch should not be needed. Please re-check if they are needed indeed and remove unnecessary ones.

Reply:
Followed the _Edit_ part of your review message. Patches were adjusted for configs in `mpconfigport.mk`. Thanks